### ISSUE TITLE
Autotag from directory name? (Import from Joplin)

### ISSUE THREAD

**Issue by ChrisJefferson** [foambubble/foam#660]
This is a two-part PR (which I could unmerge if people want)

1) I want a nice importer from Joplin (an open source markdown note storing tool) to Foam.

2) I think the simplest way to do this would be to auto-add a tag to file based on the directories it is in (so "home/roof/fix.md" would #home and #roof).

Full description:

Joplin is based around storing notes in an hierarchical structure, so export produces a nested directory structure of markdown files.

Joplin does have tags (which are stored as #tags at the top of files, so are automatically picked up by Foam), but I didn't use them much.

I wondered if Foam could auto-add a tag for files based on their directories? Personally I'd like this generally, it would mean entries in journal got `#journal` automatically, for example.

I wrote a horrible one line bash script which added a `tags:` to each file based on the directory they were in, so "home/roof/plans.md" got tags "#home" and "#roof". This (for me) made the import much nicer.

**Comment by maciejmatczak** [comment]
I would like to have directories as a first class citizen in note taking system. But you can nest folders, but not tags - tags are flat.

What about using folders as hierarchical nodes for notes? Currently I am keeping a dummy index files, so `home.md`, `root.md` that references previous file and a new note `fix.md` that would reference `root.md`. That is well reflected in a graph. I am not sure if there is already existing issue for that one.

**Comment by riccardoferretti** [comment]
Hi @maciejmatczak, a couple of notes:
1. tags can also be hierarchical (e.g. `#tag/subtag`)
2. having folders somewhat represented in the dataviz graph (I believe this is what you are referring to) could definitely be an interesting option. Whether as nodes, as you suggested, or containers. cc @jmg-duarte who is our dataviz champion

**Comment by jmg-duarte** [comment]
I agree that folders could/should be represented, I think we can provide options and expand on the topic but getting a first draft would be the priority.

**Comment by maciejmatczak** [comment]
Thanks guys, comments appreciated.

My comment here might be a bit off topic. Main "folders in graph" is #452.

BTW, "first draft" - it's implementation or functional spec / results expectation?

**Comment by riccardoferretti** [comment]
closing for inactivity
